### PR TITLE
Option to blur NSFW

### PR DESCRIPTION
**Pull request by TrollBlox**
Please give feedback on this, I know I did some stuff wrong because I have little experience with Kotlin and Jetpack Compose.

### CODE REVIEW

**Comment by TrollBlox** [comment]
It works as intended though

File: app/src/main/java/com/jerboa/Utils.kt [hunk 1]
```diff
@@ -382,8 +382,8 @@ fun prettyTimeShortener(timeString: String): String {
                 .replace(Regex("hours?"), "h")
                 .replace(Regex("days?"), "d")
                 .replace(Regex("weeks?"), "w")
-                .replace(Regex("months?"), "M")
```
Comment (twizmwazin) [inline app/src/main/java/com/jerboa/Utils.kt:385]: This change is unrelated

File: app/src/main/java/com/jerboa/ui/components/common/ImageViewerDialog.kt [hunk 2]
```diff
@@ -69,7 +69,7 @@ fun ImageViewerDialog(url: String, onBackRequest: () -> Unit) {
     }
 
     val backColor = MaterialTheme.colorScheme.scrim
-    val backColorTranslucent = MaterialTheme.colorScheme.scrim.copy(alpha = 0.4f)
+    val backColorTranslucent = MaterialTheme.colorScheme.scrim.copy(alpha = 1f)
```
Comment (twizmwazin) [inline app/src/main/java/com/jerboa/ui/components/common/ImageViewerDialog.kt:72]: Is this related to the main change?

File: app/src/main/java/com/jerboa/ui/components/post/PostListing.kt [hunk 3]
```diff
@@ -319,7 +324,7 @@ fun PostTitleAndImageLink(
         .clickable { showImageDialog = true }
     PictrsUrlImage(
         url = url,
-        nsfw = nsfwCheck(postView),
+        blur = nsfwCheck(postView) && appSettings?.blurNSFW!!,
```
Comment (twizmwazin) [inline app/src/main/java/com/jerboa/ui/components/post/PostListing.kt:327]: I think it'd be a better idea to use `?: true` instead of `!!`

File: app/src/main/java/com/jerboa/ui/components/post/PostListing.kt [hunk 4]
```diff
@@ -419,17 +429,17 @@ fun PostBody(
...
-fun PreviewStoryTitleAndMetadata() {
-    PostBody(
-        postView = samplePostView,
-        onPostLinkClick = {},
-        fullBody = false,
-        expandedImage = false,
-        account = null,
-    )
-}
+//@Preview
```
Comment (twizmwazin) [inline app/src/main/java/com/jerboa/ui/components/post/PostListing.kt:432]: Please restore this and the other previews

File: app/src/main/java/com/jerboa/ui/components/settings/SettingsActivity.kt [hunk 5]
```diff
@@ -41,16 +41,6 @@ fun SettingsActivity(
         },
         content = { padding ->
             Column(modifier = Modifier.padding(padding)) {
-                SettingsMenuLink(
```
Comment (twizmwazin) [inline app/src/main/java/com/jerboa/ui/components/settings/SettingsActivity.kt:44]: This change seems unrelated

**Review by twizmwazin — CHANGES_REQUESTED** [review]
So far in jerboa we've chose to pass specific flags into the composables rather than the whole AppSetting, so it'd be preferable to continue that. Also make sure to run `./gradlew formatKotlin`.

File: app/src/main/java/com/jerboa/Utils.kt (hunk 1 above)
Comment (TrollBlox) [reply app/src/main/java/com/jerboa/Utils.kt:385]: Sorry, I have things changed for myself and they must have gotten in here somehow. I will change it on the next commit.

**Comment by TrollBlox** [comment]
> So far in jerboa we've chose to pass specific flags into the composables rather than the whole AppSetting, so it'd be preferable to continue that.

Could you explain further?

**Comment by twizmwazin** [comment]
You've added parameters to many functions for the AppSettings object. Instead of doing this, add a parameter for blur_nsfw instead. Avoid passing the appSettings object directly.

**Comment by TrollBlox** [comment]
> You've added parameters to many functions for the AppSettings object. Instead of doing this, add a parameter for blur_nsfw instead. Avoid passing the appSettings object directly.

Oh I see what you mean now. I was already working to do that because I realized how dumb it was.

File: app/src/main/java/com/jerboa/ui/components/post/PostListing.kt (hunk 4 above)
Comment (TrollBlox) [reply app/src/main/java/com/jerboa/ui/components/post/PostListing.kt]: all of the previews should be restored.

File: app/src/main/java/com/jerboa/ui/components/settings/SettingsActivity.kt (hunk 5 above)
Comment (TrollBlox) [reply app/src/main/java/com/jerboa/ui/components/settings/SettingsActivity.kt]: i moved this back to its original location

**Comment by TrollBlox** [comment]
oops
